File: perl/1_fizz_buzz.pl
#!/usr/bin/env perl

use strict;
use warnings;

open my $fh, '<', $ARGV[0];

for my $line ( <$fh> ) {
	my @parts = split / /, $line;
	my $a     = $parts[0];
	my $b     = $parts[1];
	my $n     = $parts[2];

	my @output = ();
	for ( my $i = 1; $i <= $n; $i++ ) {
		if ( $i % $a == 0 and $i % $b == 0 ) {
			push @output, 'FB';
		}
		elsif ( $i % $a == 0 ) {
			push @output, 'F';
		}
		elsif ( $i % $b == 0 ) {
			push @output, 'B';
		}
		else {
			push @output, $i;
		}
	}

	print join( ' ', @output ) . "\n";
}

close $fh;

File: python/1_fizz_buzz.py
#!/usr/bin/env python

import sys

with open(sys.argv[1], 'r') as fh:
    lines = fh.readlines()

    for line in lines:
        line = line.strip()
        if line == '':
            continue

        (a, b, n) = [int(n) for n in line.split(' ')]

        output = []
        for i in range(1, n + 1):
            if i % a == 0 and i % b == 0:
                output.append('FB')
            elif i % a == 0:
                output.append('F')
            elif i % b == 0:
                output.append('B')
            else:
                output.append(str(i))

        print ' '.join(output)

